raster-foundry/raster-foundry
# Support STAC exports on annotation projects

## Overview
* Remove references to RF projects
* Align annotations / label items with STAC spec

### Checklist

- ~~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~~

### Demo

### Notes
Single label class group projects should be given a default label class group named something sensible such as "label" to align with the old stac exports.

## Testing Instructions

- load the dev database
- run migrations until they error
- run the batch command to migrate annotation projects:
```./scripts/console batch 'java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main liberate-annotation-projects http://localhost:8081'```
- run migrations again
- change the name of the existing label class group to "label":
```
update annotation_label_class_groups set name = 'label';
```
- Give your user `groundworkUser` scope permissions
``` 
update users set scopes = '{"platforms:admin;groundworkUser"}' where email = 'dev@rasterfoundry.com';
```
- create a stac export 
```
insert into stac_exports (id, created_at, created_by, modified_at, owner, name, export_status, task_statuses, annotation_project_id) values (uuid_generate_v4(), now(), 'auth0|59318a9d2fbbca3e16bcfc92', now(), 'auth0|59318a9d2fbbca3e16bcfc92', 'annotation label export', 'TOBEEXPORTED', '{"LABELLED"}', '84db6e9f-a7d4-4f06-90c0-184e35965f72');
```
You'll probably need to change the annotation project id to the random one selected on your machine
- Build your batch jar & container
```
cd app-backend && ./sbt batch/assembly && docker-compose build batch
```
- start the API server etc with `./scripts/server`
- run the batch command
```
sudo -E ./scripts/console batch 'java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main write_stac_catalog "77e9f319-a959-439c-a7fd-6a193a98c373"'
```
(replace id with your stac export's ID)
- use the logged s3 link and fetch it.
```
aws s3 sync <path> ./stac-exports/
```
- Verify that the resulting stac export looks good. Of note:
  - The stac item properties `label:classes` and `label:property` should contain all the valid annotation classes
  - the annotation classes on the geojson itself. I inspected them using `jq ".features[0]" data.geojson` to look at the first label

Closes #5287
